ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -63,6 +67,18 @@ export default class PermissionPageContainerContent extends PureComponent {
     const { origin, subjectType } = subjectMetadata;
     const displayOrigin =
       subjectType === SubjectType.Website ? getURLHost(origin) : origin;
+
+    // permissionDiffMap is expected to be present when an incremental permission request is made
+    // This occurs when a "wallet_switchEthereumChain" request comes in and we already have a set of permissions
+    const permissionDiffMap = request.diff?.permissionDiffMap;
+    // Extract the requested chain IDs from the permission diff, specifically from the CAIP-25 endowment permission
+    // This represents the new chains being requested in addition to existing permissions
+    const permissionDiffRequestedChainIds = getPermittedEthChainIds(
+      permissionDiffMap?.[Caip25EndowmentPermissionName]?.authorizedScopes ?? {
```

Comment:
authorizedScopes should come from a the caveat type constant https://github.com/MetaMask/core/blob/main/packages/chain-agnostic-permission/src/caip25Permission.ts#L56